sdk/core/azure-json/src/main/java/com/azure/json/JsonBuilder.java

```diff
@@ -0,0 +1,148 @@
+package com.azure.json;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+
+import static com.azure.json.JsonToken.END_DOCUMENT;
+
+/**
+ * probably unnecessary, but I'm leaving it in for now
+ * The thinking was should we consider hiding all the elements in the implementation
+ * and only exposing the JsonBuilder class to the user? It may be good to prevent people
+ * making random JsonElements without a proper container like Object or Array,
+ * but it may also make it harder to use.
+ */
+public class JsonBuilder {
+    /**
+     *
+     * @return
+     */
+    public JsonArray createArray() {
+        return new JsonArray();
+    }
+
+    /**
+     *
+     * @return
+     */
+    public JsonObject createObject() {
+        return new JsonObject();
+    }
+
+    /**
+     * Deserialize a JSON string into a JsonElement. There are overloads for byte[] arrays, InputStreams, and Readers
+     * these are all valid dependency injections for JsonReader
+     * @param json is a valid JSON string
+     * @return a JsonElement that will contain an object or array with properties, and nested structures
+     * ready to interact with
+     * @throws IOException if the JSON string is invalid
+     */
+    public JsonElement deserialize(String json) throws IOException {
+        return this.buildOutput(JsonProviders.createReader(json));
+    }
+
+    /**
+     *
+     * @param json
+     * @return
+     * @throws IOException
+     */
+    public JsonElement deserialize(byte[] json) throws IOException {
+        return this.buildOutput(JsonProviders.createReader(json));
+    }
+
+    /**
+     *
+     * @param json
+     * @return
+     * @throws IOException
+     */
+    public JsonElement deserialize(InputStream json) throws IOException {
+        return this.buildOutput(JsonProviders.createReader(json));
+    }
+
+    /**
+     *
+     * @param json
+     * @return
+     * @throws IOException
+     */
+    public JsonElement deserialize(Reader json) throws IOException {
+        return this.buildOutput(JsonProviders.createReader(json));
+    }
```

Comment:
For an initial release, I think we could just provide `buildOutput` as these are all just convenience methods for that API

Reply:
@Robsimpson rework so that it is just buildoutput, without the wrappers.